sdcm/utils/common.py

```diff
@@ -242,14 +243,39 @@ def download_file(self, link, dst_dir):
 
 
 def get_latest_gemini_version():
+    """search latest version of gemini
+
+    Get list of gemini files from s3:
+      ['gemini/0.9.1/checksums.txt',
+       'gemini/0.9.1/gemini_0.9.1_Linux_x86_64.tar.gz',
+       'gemini/0.9.2/checksums.txt',
+       ...
+       'gemini/1.3.1/gemini_1.3.1_Linux_x86_64.tar.gz',]
+       ...
+    Split each row by / and get version '0.9.2'
+    Convert this version to [1, 9, 2] and add to list of version
+        [
+            [0, 9, 2],
+            [1, 6, 9],
+            [1, 6, 10]
+        ]
+    Sort list with version by each element in version and return latest
+    """
     bucket_name = 'downloads.scylladb.com'
 
     results = S3Storage(bucket_name).search_by_path(path='gemini')
-    versions = set()
+    versions = []
     for result_file in results:
-        versions.add(result_file.split('/')[1])
+        gemini_version = [int(num) for num in result_file.split('/')[1].split('.')]
```

Comment:
I wouldn't recommend handling versions like that, or assuming they are always numbers
see
https://stackoverflow.com/a/11887885/459189

and then you can just call sort of the list you get

Reply:
Thanks!

Reply:
fixed